src/MinimalDelegation.sol

```diff
@@ -151,26 +151,31 @@ contract MinimalDelegation is
 
         Settings settings = keySettings[keyHash];
         if (settings.isExpired()) revert IKeyManagement.KeyExpired();
+        uint40 expiry = settings.expiration();
 
         IHook hook = settings.hook();
         validationData = hook.hasPermission(HooksLib.VALIDATE_USER_OP_FLAG)
             ? hook.validateUserOp(keyHash, userOp, userOpHash)
```

Comment:
should we pass expiry in here to the hook or is it something we should only use internally?

Reply:
imo the hook can query the contract to check expiry since it has the keyHash